### PR TITLE
Cap string variable value length and unify corresponding SQL columns size

### PR DESCRIPTION
**Pull request by Kenpachi2k13**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Currently the maximum length of string variable values is inconsistent.
 * If a variable is not stored in a SQL table its value's length is not restricted.
 * If a variable is stored in `acc_reg_str_db`, `char_reg_str_db` or `global_acc_reg_str_db` table its value's length is restricted to 254.
 * If a variable is stored in `mapreg` table its value's length is restricted to 255.

Furthermore some fields are too small to use the maximum supported value length.

To have a consistent size I capped the length of all string variable values to 255 and unified the corresponding SQL columns size accordingly.

**Issues addressed:** #1037

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: sql-files/upgrades/2020-05-01--04-44.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU General Public License for more details.
+--
+-- You should have received a copy of the GNU General Public License
+-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ALTER TABLE `acc_reg_str_db` MODIFY `value` VARCHAR(255) NOT NULL DEFAULT '';
```
Comment (4144) [inline sql-files/upgrades/2020-05-01--04-44.sql:21]: default value look like different with defaults from main.sql

File: src/map/atcommand.c [hunk 2]
```diff
@@ -8945,13 +8945,17 @@ ACMD(accinfo)
...
+	char reg[SCRIPT_VARNAME_LENGTH + 1];
+	char val[SCRIPT_STRING_VAR_LENGTH + 1];
 	struct script_data* data;
 	int toset = 0;
 	bool is_str = false;
 	size_t len;
 
-	if (!*message || (toset = sscanf(message, "%32s %253[^\n]", reg, val)) < 1) {
+	char format[20];
+	snprintf(format, sizeof(format), "%%%ds %%%d[^\n]", SCRIPT_VARNAME_LENGTH, SCRIPT_STRING_VAR_LENGTH);
```
Comment (4144) [inline src/map/atcommand.c:8956]: better use safesnprintf

Comment (4144) [reply src/map/atcommand.c:8956]: also if SCRIPT_VARNAME_LENGTH will be 2^32, will be buffer overflow

Comment (Kenpachi2k13) [reply src/map/atcommand.c:8956]: Why should someone set `SCRIPT_VARNAME_LENGTH` to 4294967296? o.O

File: sql-files/upgrades/2020-05-01--04-44.sql (hunk 1 above)
Comment (Kenpachi2k13) [reply sql-files/upgrades/2020-05-01--04-44.sql]: Fixed.

File: src/map/atcommand.c (hunk 2 above)
Comment (Kenpachi2k13) [reply src/map/atcommand.c]: > better use safesnprintf
Fixed.

> also if SCRIPT_VARNAME_LENGTH will be 2^32, will be buffer overflow
Added static assert check for SCRIPT_VARNAME_LENGTH.

File: src/map/script.c [hunk 3]
```diff
@@ -3439,43 +3439,53 @@ static int set_reg(struct script_state *st, struct map_session_data *sd, int64 n
 	if (is_string_variable(name)) {// string variable
 		const char *str = (const char*)value;
 
+		if (strlen(str) > SCRIPT_STRING_VAR_LENGTH) {
```
Comment (MishimaHaruna) [inline src/map/script.c:3442]: How about only limiting the variables that are persisted to the database? Their use in scripts is (or, should) be far less common than the non-persisted variables, and unless there's a compelling reason to limit the maximum string length I'd avoid it at all costs.

Comment (Kenpachi2k13) [reply src/map/script.c:3442]: Sure, easy to do... But I thought you're a friend of having consistency. That's why I made no exception for temporary variables.

Comment (MishimaHaruna) [reply src/map/script.c]: I agree with the consistency part, but it's also true that most use cases for string variables only care about scope variables, and are never saved to the database.

I'm mostly thinking of long menu sequences constructed by concatenating options into a string (which the client limits to 127 options if I remember correctly, and we surely want each of them to be able to be more than 1 character long, not including the `:`), or of long text to print through mes (I believe the client can support up to 4k characters in a single message dialog, with some NEMO patches applied?), so it makes sense not to add an artificial restriction on those strings unless it's necessary for technical reasons

**Pull request merged**